OpenRA.Game/Graphics/SpriteCache.cs

```diff
@@ -49,30 +50,34 @@ public sealed class SpriteCache : IDisposable
 		}
 
 		public int ReserveSprites(string filename, IEnumerable<int> frames, MiniYamlNode.SourceLocation location,
-			Func<ISpriteFrame, ISpriteFrame> adjustFrame = null, bool premultiplied = false)
+			AdjustFrame adjustFrame = null, bool premultiplied = false)
 		{
 			var token = nextReservationToken++;
 			spriteReservations[token] = (frames?.ToArray(), location, adjustFrame, premultiplied);
 			reservationsByFilename.GetOrAdd(filename, _ => new List<int>()).Add(token);
 			return token;
 		}
 
-		static ISpriteFrame[] GetFrames(IReadOnlyFileSystem fileSystem, string filename, ISpriteLoader[] loaders, out TypeDictionary metadata)
+		static ISpriteFrame[] GetFrames(IReadOnlyFileSystem fileSystem, string filename, ISpriteLoader[] loaders)
```

Comment:
while this probably isn't the right place to get metadata, it seems to be the only place. I recall @IceReaper having issues about metadata being disposed

Reply:
Yeah, kknd has per frame metadata like 2d bone positions. Thats required to make the actors look like they should look when animating them